haystack/management/commands/build_solr_schema.py

```diff
@@ -22,15 +28,66 @@ def add_arguments(self, parser):
             "-u", "--using", default=constants.DEFAULT_ALIAS,
             help='If provided, chooses a connection to work with.'
         )
+        parser.add_argument(
+            "-c", "--configure_dir",
+            help='If provided, attempts to configure a core located in the given directory by removing the managed-schema.xml(renaming), configuring the core to use a classic (non-dynamic) schema, and generating the schema.xml from the template provided in'
+        )
+        parser.add_argument(
+            "-r", "--reload",
+            help='If provided, attempts to automatically reload the solr core'
+        )
+
 
     def handle(self, **options):
         """Generates a Solr schema that reflects the indexes."""
         using = options.get('using')
-        schema_xml = self.build_template(using=using)
+        if not isinstance(connections[using].get_backend(), SolrSearchBackend):
+            raise ImproperlyConfigured("'%s' isn't configured as a SolrEngine)." % connections[using].get_backend().connection_alias)
+
+        schema_xml = self.build_template(using=using,tfile=Command.schema_template_loc)
+        solrcfg_xml = self.build_template(using=using,tfile=Command.solrcfg_template_loc)
 
         if options.get('filename'):
+            self.stdout.write("Trying to write schema file located at {}".format(options.get('filename')))
             self.write_file(options.get('filename'), schema_xml)
-        else:
+            if options.get('reload'):
+                connections[using].get_backend().reload()
+
+        if options.get('configure_dir'):
+            cdir = options.get('configure_dir')
+            self.stdout.write("Trying to configure core located at {}".format(cdir))
+            if os.path.isfile(cdir+'/managed-schema'):
```

Comment:
We should use `os.path.join` for all of the filesystem path construction logic in this section to avoid issues with Windows